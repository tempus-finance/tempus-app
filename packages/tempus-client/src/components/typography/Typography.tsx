import { CSSProperties, FC } from 'react';

type TypographyVariant = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'body-text' | 'disclaimer-text';
type TypographyColor = 'default' | 'accent' | 'inverted';

const typographyStyleMap = new Map<TypographyVariant, CSSProperties>();
typographyStyleMap.set('h1', {
  fontFamily: "'Inter', sans-serif",
  fontWeight: 900,
  fontSize: '36px',
  fontStyle: 'normal',
  lineHeight: '44px',
});
typographyStyleMap.set('h2', {
  fontFamily: "'Inter', sans-serif",
  fontWeight: 900,
  fontSize: '26px',
  fontStyle: 'normal',
  lineHeight: '31px',
});
typographyStyleMap.set('h3', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 500,
  fontSize: '22px',
  fontStyle: 'normal',
  lineHeight: '26px',
});
typographyStyleMap.set('h4', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 700,
  fontSize: '18px',
  fontStyle: 'normal',
  lineHeight: '21px',
});
typographyStyleMap.set('h5', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 500,
  fontSize: '16px',
  fontStyle: 'normal',
  lineHeight: '19px',
});
typographyStyleMap.set('body-text', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 400,
  fontSize: '16px',
  fontStyle: 'normal',
  lineHeight: '19px',
});
typographyStyleMap.set('disclaimer-text', {
  fontFamily: "'Roboto', sans-serif",
  fontWeight: 400,
  fontSize: '12px',
  fontStyle: 'normal',
  lineHeight: '14px',
});

interface TypographyProps {
  variant: TypographyVariant;
  color: TypographyColor;
}

const Typography: FC<TypographyProps> = props => {
  let color: string;
  switch (props.color) {
    case 'default':
      color = '#222222';
      break;
    case 'accent':
      color = '#F5AC37';
      break;
    case 'inverted':
      color = '#FFFFFF';
      break;
  }

  return (
    <div
      style={{
        ...typographyStyleMap.get(props.variant),
        color: color,
      }}
    >
      {props.children}
    </div>
  );
};
export default Typography;
