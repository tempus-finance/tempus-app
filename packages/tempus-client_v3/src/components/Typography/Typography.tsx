import React, { CSSProperties, FC } from 'react';
import parse from 'html-react-parser';

type TypographyVariant =
  | 'header'
  | 'subheader'
  | 'title'
  | 'subtitle'
  | 'body-primary'
  | 'body-secondary'
  | 'boyd-tertiary';

type TypographyColor = 'text-primary';

type TypographyWeight = 'regular' | 'medium' | 'bold';

type TypographySpacing = 'regular' | 'mono';

const typographyVariantMap = new Map<TypographyVariant, CSSProperties>();
typographyVariantMap.set('header', {
  fontStyle: 'normal',
  fontSize: '40px',
  lineHeight: '48px',
});
typographyVariantMap.set('subheader', {
  fontStyle: 'normal',
  fontSize: '32px',
  lineHeight: '40px',
});
typographyVariantMap.set('title', {
  fontStyle: 'normal',
  fontSize: '24px',
  lineHeight: '32px',
});
typographyVariantMap.set('subtitle', {
  fontStyle: 'normal',
  fontSize: '20px',
  lineHeight: '28px',
});
typographyVariantMap.set('body-primary', {
  fontStyle: 'normal',
  fontSize: '16px',
  lineHeight: '24px',
});
typographyVariantMap.set('body-secondary', {
  fontStyle: 'normal',
  fontSize: '12px',
  lineHeight: '16px',
});
typographyVariantMap.set('boyd-tertiary', {
  fontStyle: 'normal',
  fontSize: '10px',
  lineHeight: '16px',
});

const typographyColorMap = new Map<TypographyColor, string>();
typographyColorMap.set('text-primary', '#222222');

const typographyWeightMap = new Map<TypographyWeight, number>();
typographyWeightMap.set('regular', 400);
typographyWeightMap.set('medium', 500);
typographyWeightMap.set('bold', 700);

const typographySpacingMap = new Map<TypographySpacing, string>();
typographySpacingMap.set('regular', "'DM Sans', sans-serif");
typographySpacingMap.set('mono', "'Azeret Mono', monospace");

interface TypographyProps {
  variant: TypographyVariant;
  color?: TypographyColor;
  weight?: TypographyWeight;
  spacing?: TypographySpacing;
  html?: string;
}

const Typography: FC<TypographyProps & React.HTMLProps<HTMLDivElement>> = props => {
  const { variant, color, weight, spacing, html, children } = props;

  let fontColor: string | undefined;
  if (color) {
    fontColor = typographyColorMap.get(color);
  } else {
    fontColor = typographyColorMap.get('text-primary');
  }

  let fontWeight: number | undefined;
  if (weight) {
    fontWeight = typographyWeightMap.get(weight);
  } else {
    fontWeight = typographyWeightMap.get('regular');
  }

  let fontFamily: string | undefined;
  if (spacing) {
    fontFamily = typographySpacingMap.get(spacing);
  } else {
    fontFamily = typographySpacingMap.get('regular');
  }

  return (
    <div
      style={{
        ...typographyVariantMap.get(variant),
        color: fontColor,
        fontWeight: fontWeight,
        fontFamily: fontFamily,
      }}
    >
      {html ? parse(html) : children}
    </div>
  );
};
export default Typography;
