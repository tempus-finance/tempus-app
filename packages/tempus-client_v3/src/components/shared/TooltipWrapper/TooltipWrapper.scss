.tc__tooltip-wrapper {
  position: relative;

  // the CSS variable are here for React components to override
  --anchorPosX: 0px;
  --anchorPosY: 0px;

  .tc__tooltip-wrapper-backdrop {
    position: absolute;
    left: calc(-1 * var(--anchorPosX));
    top: calc(-1 * var(--anchorPosY));
    width: 100vw;
    height: 100vh;
    z-index: 9997;
  }

  .tc__tooltip-wrapper-anchor {
    position: inherit;
    z-index: 9998;
  }

  .tc__tooltip {
    z-index: 9999;
  }
}

.tc__tooltip-wrapper-anchor {
  display: flex;
}
