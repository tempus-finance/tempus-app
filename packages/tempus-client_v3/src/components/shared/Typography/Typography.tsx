import React, { CSSProperties, FC } from 'react';
import parse from 'html-react-parser';
import { colors } from '../Color';

type TypographyVariant =
  | 'header'
  | 'subheader'
  | 'title'
  | 'subtitle'
  | 'body-primary'
  | 'body-secondary'
  | 'body-tertiary';

export type TypographyColor =
  | 'text-primary'
  | 'text-inverted'
  | 'text-success'
  | 'text-disabled'
  | 'text-disabled-secondary';

export type TypographyWeight = 'regular' | 'medium' | 'bold';

type TypographyType = 'regular' | 'mono';

const typographyVariantMap = new Map<TypographyVariant, CSSProperties>();
typographyVariantMap.set('header', {
  fontStyle: 'normal',
  fontSize: '40px',
  lineHeight: '48px',
});
typographyVariantMap.set('subheader', {
  fontStyle: 'normal',
  fontSize: '32px',
  lineHeight: '40px',
});
typographyVariantMap.set('title', {
  fontStyle: 'normal',
  fontSize: '24px',
  lineHeight: '32px',
});
typographyVariantMap.set('subtitle', {
  fontStyle: 'normal',
  fontSize: '20px',
  lineHeight: '28px',
});
typographyVariantMap.set('body-primary', {
  fontStyle: 'normal',
  fontSize: '16px',
  lineHeight: '24px',
});
typographyVariantMap.set('body-secondary', {
  fontStyle: 'normal',
  fontSize: '12px',
  lineHeight: '16px',
});
typographyVariantMap.set('body-tertiary', {
  fontStyle: 'normal',
  fontSize: '10px',
  lineHeight: '16px',
});

const typographyColorMap = new Map<TypographyColor, string>();
typographyColorMap.set('text-primary', colors.textPrimary);
typographyColorMap.set('text-inverted', colors.textInverted);
typographyColorMap.set('text-success', colors.textSuccess);
typographyColorMap.set('text-disabled', colors.textDisabled);
typographyColorMap.set('text-disabled-secondary', colors.textDisabledSecondary);

const typographyWeightMap = new Map<TypographyWeight, number>();
typographyWeightMap.set('regular', 400);
typographyWeightMap.set('medium', 500);
typographyWeightMap.set('bold', 700);

const typographyTypeMap = new Map<TypographyType, string>();
typographyTypeMap.set('regular', "'DM Sans', sans-serif");
typographyTypeMap.set('mono', "'Azeret Mono', monospace");

interface TypographyProps {
  variant: TypographyVariant;
  color?: TypographyColor;
  opacity?: number;
  weight?: TypographyWeight;
  type?: TypographyType;
  html?: string;
}

const Typography: FC<TypographyProps & React.HTMLProps<HTMLDivElement>> = props => {
  const { variant, color, weight, opacity = 1, type, html, children } = props;

  let fontColor: string | undefined;
  if (color) {
    fontColor = typographyColorMap.get(color);
  } else {
    fontColor = typographyColorMap.get('text-primary');
  }

  let fontWeight: number | undefined;
  if (weight) {
    fontWeight = typographyWeightMap.get(weight);
  } else {
    fontWeight = typographyWeightMap.get('regular');
  }

  let fontFamily: string | undefined;
  if (type) {
    fontFamily = typographyTypeMap.get(type);
  } else {
    fontFamily = typographyTypeMap.get('regular');
  }

  return (
    <div
      style={{
        ...typographyVariantMap.get(variant),
        color: fontColor,
        fontWeight: fontWeight,
        fontFamily: fontFamily,
        opacity,
      }}
    >
      {html ? parse(html) : children}
    </div>
  );
};
export default Typography;
