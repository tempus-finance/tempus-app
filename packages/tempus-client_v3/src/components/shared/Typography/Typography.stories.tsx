import React, { CSSProperties } from 'react';

import Typography from './Typography';

export default {
  title: 'Typography',
  component: Typography,
  argTypes: {},
};

const style: CSSProperties = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

export const Header = () => (
  <div style={style}>
    <Typography variant="header">Header</Typography>
    <br />
    <Typography variant="header" weight="medium">
      Header Medium
    </Typography>
    <br />
    <Typography variant="header" weight="bold">
      Header Bold
    </Typography>
    <br />
    <Typography variant="header" type="mono">
      Header Mono
    </Typography>
    <br />
    <Typography variant="header" weight="medium" type="mono">
      Header Mono Medium
    </Typography>
    <br />
    <Typography variant="header" weight="bold" type="mono">
      Header Mono Bold
    </Typography>
  </div>
);

export const Subheader = () => (
  <div style={style}>
    <Typography variant="subheader">Subheader</Typography>
    <br />
    <Typography variant="subheader" weight="medium">
      Subheader Medium
    </Typography>
    <br />
    <Typography variant="subheader" weight="bold">
      Subheader Bold
    </Typography>
    <br />
    <Typography variant="subheader" type="mono">
      Subheader Mono
    </Typography>
    <br />
    <Typography variant="subheader" weight="medium" type="mono">
      Subheader Mono Medium
    </Typography>
    <br />
    <Typography variant="subheader" weight="bold" type="mono">
      Subheader Mono Bold
    </Typography>
  </div>
);

export const Title = () => (
  <div style={style}>
    <Typography variant="title">Title</Typography>
    <br />
    <Typography variant="title" weight="medium">
      Title Medium
    </Typography>
    <br />
    <Typography variant="title" weight="bold">
      Title Bold
    </Typography>
    <br />
    <Typography variant="title" type="mono">
      Title Mono
    </Typography>
    <br />
    <Typography variant="title" weight="medium" type="mono">
      Title Mono Medium
    </Typography>
    <br />
    <Typography variant="title" weight="bold" type="mono">
      Title Mono Bold
    </Typography>
  </div>
);

export const Subtitle = () => (
  <div style={style}>
    <Typography variant="subtitle">Subtitle</Typography>
    <br />
    <Typography variant="subtitle" weight="medium">
      Subtitle Medium
    </Typography>
    <br />
    <Typography variant="subtitle" weight="bold">
      Subtitle Bold
    </Typography>
    <br />
    <Typography variant="subtitle" type="mono">
      Subtitle Mono
    </Typography>
    <br />
    <Typography variant="subtitle" weight="medium" type="mono">
      Subtitle Mono Medium
    </Typography>
    <br />
    <Typography variant="subtitle" weight="bold" type="mono">
      Subtitle Mono Bold
    </Typography>
  </div>
);

export const BodyPrimary = () => (
  <div style={style}>
    <Typography variant="body-primary">Body Primary</Typography>
    <br />
    <Typography variant="body-primary" weight="medium">
      Body Primary Medium
    </Typography>
    <br />
    <Typography variant="body-primary" weight="bold">
      Body Primary Bold
    </Typography>
    <br />
    <Typography variant="body-primary" type="mono">
      Body Primary Mono
    </Typography>
    <br />
    <Typography variant="body-primary" weight="medium" type="mono">
      Body Primary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-primary" weight="bold" type="mono">
      Body Primary Mono Bold
    </Typography>
  </div>
);

export const BodySecondary = () => (
  <div style={style}>
    <Typography variant="body-secondary">Body Secondary</Typography>
    <br />
    <Typography variant="body-secondary" weight="medium">
      Body Secondary Medium
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="bold">
      Body Secondary Bold
    </Typography>
    <br />
    <Typography variant="body-secondary" type="mono">
      Body Secondary Mono
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="medium" type="mono">
      Body Secondary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-secondary" weight="bold" type="mono">
      Body Secondary Mono Bold
    </Typography>
  </div>
);

export const BodyTertiary = () => (
  <div style={style}>
    <Typography variant="body-tertiary">Body Tertiary</Typography>
    <br />
    <Typography variant="body-tertiary" weight="medium">
      Body Tertiary Medium
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="bold">
      Body Tertiary Bold
    </Typography>
    <br />
    <Typography variant="body-tertiary" type="mono">
      Body Tertiary Mono
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="medium" type="mono">
      Body Tertiary Mono Medium
    </Typography>
    <br />
    <Typography variant="body-tertiary" weight="bold" type="mono">
      Body Tertiary Mono Bold
    </Typography>
  </div>
);
