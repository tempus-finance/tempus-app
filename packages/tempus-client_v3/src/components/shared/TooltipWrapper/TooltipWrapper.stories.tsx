import { ComponentStory, ComponentMeta } from '@storybook/react';
import Typography from '../Typography';

import TooltipWrapper from './TooltipWrapper';

export default {
  title: 'TooltipWrapper',
  component: TooltipWrapper,
  argTypes: {},
} as ComponentMeta<typeof TooltipWrapper>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof TooltipWrapper> = args => (
  <div style={style}>
    <TooltipWrapper {...args}>
      <Typography variant="title">anchor</Typography>
    </TooltipWrapper>
  </div>
);

export const TooltipWrapperSimple = Template.bind({});
TooltipWrapperSimple.args = {
  tooltipContent: <Typography variant="body-primary">Tooltip</Typography>,
};

export const TooltipWrapperClickToOpen = Template.bind({});
TooltipWrapperClickToOpen.args = {
  tooltipContent: <Typography variant="body-primary">Tooltip</Typography>,
  openEvent: 'click',
};

export const TooltipWrapperHoverToOpen = Template.bind({});
TooltipWrapperHoverToOpen.args = {
  tooltipContent: <Typography variant="body-primary">Tooltip</Typography>,
  openEvent: 'mouseover',
};

export const TooltipWrapperBottomLeft = Template.bind({});
TooltipWrapperBottomLeft.args = {
  tooltipContent: <div style={{ width: '320px', height: '240px' }}>Tooltip</div>,
  placement: 'bottom-left',
};

export const TooltipWrapperBottomCenter = Template.bind({});
TooltipWrapperBottomCenter.args = {
  tooltipContent: <div style={{ width: '320px', height: '240px' }}>Tooltip</div>,
  placement: 'bottom-center',
};

export const TooltipWrapperBottomRight = Template.bind({});
TooltipWrapperBottomRight.args = {
  tooltipContent: <div style={{ width: '320px', height: '240px' }}>Tooltip</div>,
  placement: 'bottom-right',
};
