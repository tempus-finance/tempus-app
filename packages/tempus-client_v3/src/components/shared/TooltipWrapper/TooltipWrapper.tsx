import { FC, HTMLProps, ReactNode, useCallback, useMemo, useState } from 'react';
import ButtonWrapper, { ButtonWrapperProps } from '../ButtonWrapper/ButtonWrapper';
import Tooltip, { TooltipPlacement } from '../Tooltip/Tooltip';

import './TooltipWrapper.scss';

export type OpenEvent = 'click' | 'mouseover';

export interface TooltipWrapperProps {
  tooltipContent: ReactNode;
  placement?: TooltipPlacement;
  openEvent?: OpenEvent;
  onOpen?: () => void;
  onClose?: () => void;
}

const TooltipWrapper: FC<TooltipWrapperProps> = props => {
  const { tooltipContent, placement, openEvent = 'click', onOpen, onClose, children } = props;

  const [open, setOpen] = useState<boolean>(false);

  const handleOpen = useCallback(() => {
    setOpen(true);
    onOpen?.();
  }, [onOpen]);
  const handleClose = useCallback(() => {
    setOpen(false);
    onClose?.();
  }, [onClose]);
  const buttonProps = useMemo<ButtonWrapperProps>(() => {
    const obj: ButtonWrapperProps = {};
    switch (openEvent) {
      case 'click':
      default:
        obj.onClick = handleOpen;
        break;
      case 'mouseover':
        obj.onMouseOver = handleOpen;
        obj.onFocus = handleOpen;
        break;
    }

    return obj;
  }, [openEvent, handleOpen]);
  const backdropProps = useMemo<HTMLProps<HTMLDivElement>>(() => {
    const obj: HTMLProps<HTMLDivElement> = {};
    switch (openEvent) {
      case 'click':
      default:
        obj.onClick = handleClose;
        break;
      case 'mouseover':
        obj.onMouseOver = handleClose;
        obj.onFocus = handleClose;
        break;
    }

    return obj;
  }, [openEvent, handleClose]);

  if (!open) {
    return <ButtonWrapper {...buttonProps}>{children}</ButtonWrapper>;
  }

  return (
    <div className="tc__tooltip-wrapper">
      <div className="tc__tooltip-wrapper-backdrop" {...backdropProps} />
      <ButtonWrapper className="tc__tooltip-wrapper-anchor">{children}</ButtonWrapper>
      <Tooltip open placement={placement}>
        {tooltipContent}
      </Tooltip>
    </div>
  );
};

export default TooltipWrapper;
