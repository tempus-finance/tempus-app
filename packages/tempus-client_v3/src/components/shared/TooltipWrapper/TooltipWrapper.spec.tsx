import { fireEvent, render } from '@testing-library/react';
import Typography from '../Typography';
import TooltipWrapper, { OpenEvent, TooltipWrapperProps } from './TooltipWrapper';

const mockOnOpen = jest.fn();
const mockOnClose = jest.fn();

const tooltipText = 'Tooltip';
const anchorText = 'Anchor';

const defaultProps: TooltipWrapperProps = {
  tooltipContent: <Typography variant="body-primary">{tooltipText}</Typography>,
  onOpen: mockOnOpen,
  onClose: mockOnClose,
};

const subject = (props: TooltipWrapperProps) =>
  render(
    <TooltipWrapper {...props}>
      <Typography variant="body-primary">{anchorText}</Typography>
    </TooltipWrapper>,
  );

describe('TooltipWrapper', () => {
  it('render a simple TooltipWrapper', () => {
    const { container, getByRole, queryByText } = subject(defaultProps);

    const anchor = getByRole('button');
    const actualTooltipText = queryByText(tooltipText);
    const actualAnchorText = queryByText(anchorText);

    expect(container).not.toBeNull();
    expect(anchor).not.toBeNull();
    expect(actualTooltipText).toBeNull();
    expect(actualAnchorText).toHaveTextContent(anchorText);
    expect(mockOnOpen).not.toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    expect(container).toMatchSnapshot();
  });

  it('render a simple TooltipWrapper, click to open tooltip', () => {
    const props = { ...defaultProps, openEvent: 'click' as OpenEvent };
    const { container, getByRole, queryByText } = subject(props);

    const anchor = getByRole('button');

    expect(container).not.toBeNull();
    expect(anchor).not.toBeNull();
    expect(mockOnOpen).not.toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    fireEvent.click(anchor);

    const actualTooltipText = queryByText(tooltipText);
    const actualAnchorText = queryByText(anchorText);

    expect(actualTooltipText).toHaveTextContent(tooltipText);
    expect(actualAnchorText).toHaveTextContent(anchorText);
    expect(mockOnOpen).toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    expect(container).toMatchSnapshot();
  });

  it('render a simple TooltipWrapper, click to open tooltip, click backdrop to close', () => {
    const props = { ...defaultProps, openEvent: 'click' as OpenEvent };
    const { container, getByRole, queryByText } = subject(props);

    const anchor = getByRole('button');

    expect(container).not.toBeNull();
    expect(anchor).not.toBeNull();
    expect(mockOnOpen).not.toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    fireEvent.click(anchor);

    const actualTooltipText1 = queryByText(tooltipText);
    const actualAnchorText1 = queryByText(anchorText);
    const backdrop = container.querySelector('.tc__tooltip-wrapper-backdrop');

    expect(actualTooltipText1).toHaveTextContent(tooltipText);
    expect(actualAnchorText1).toHaveTextContent(anchorText);
    expect(backdrop).not.toBeNull();
    expect(mockOnOpen).toHaveBeenCalledTimes(1);
    expect(mockOnClose).toHaveBeenCalledTimes(0);

    fireEvent.click(backdrop as Element);

    const actualTooltipText2 = queryByText(tooltipText);
    const actualAnchorText2 = queryByText(anchorText);

    expect(actualTooltipText2).toBeNull();
    expect(actualAnchorText2).toHaveTextContent(anchorText);
    expect(mockOnOpen).toHaveBeenCalledTimes(1);
    expect(mockOnClose).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });

  it('render a simple TooltipWrapper, mouseover to open tooltip', () => {
    const props = { ...defaultProps, openEvent: 'mouseover' as OpenEvent };
    const { container, getByRole, queryByText } = subject(props);

    const anchor = getByRole('button');

    expect(container).not.toBeNull();
    expect(anchor).not.toBeNull();
    expect(mockOnOpen).not.toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    fireEvent.mouseOver(anchor);

    const actualTooltipText = queryByText(tooltipText);
    const actualAnchorText = queryByText(anchorText);

    expect(actualTooltipText).toHaveTextContent(tooltipText);
    expect(actualAnchorText).toHaveTextContent(anchorText);
    expect(mockOnOpen).toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    expect(container).toMatchSnapshot();
  });

  it('render a simple TooltipWrapper, mouseover to open tooltip, mpuseout to close', () => {
    const props = { ...defaultProps, openEvent: 'mouseover' as OpenEvent };
    const { container, getByRole, queryByText } = subject(props);

    const anchor = getByRole('button');

    expect(container).not.toBeNull();
    expect(anchor).not.toBeNull();
    expect(mockOnOpen).not.toHaveBeenCalled();
    expect(mockOnClose).not.toHaveBeenCalled();

    fireEvent.mouseOver(anchor);

    const actualTooltipText1 = queryByText(tooltipText);
    const actualAnchorText1 = queryByText(anchorText);
    const backdrop = container.querySelector('.tc__tooltip-wrapper-backdrop');

    expect(actualTooltipText1).toHaveTextContent(tooltipText);
    expect(actualAnchorText1).toHaveTextContent(anchorText);
    expect(backdrop).not.toBeNull();
    expect(mockOnOpen).toHaveBeenCalledTimes(1);
    expect(mockOnClose).toHaveBeenCalledTimes(0);

    fireEvent.mouseOver(backdrop as Element);

    const actualTooltipText2 = queryByText(tooltipText);
    const actualAnchorText2 = queryByText(anchorText);

    expect(actualTooltipText2).toBeNull();
    expect(actualAnchorText2).toHaveTextContent(anchorText);
    expect(mockOnOpen).toHaveBeenCalledTimes(1);
    expect(mockOnClose).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });
});
