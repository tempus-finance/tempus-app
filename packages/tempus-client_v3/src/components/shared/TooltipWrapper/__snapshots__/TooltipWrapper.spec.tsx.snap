// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipWrapper render a simple TooltipWrapper 1`] = `
<div>
  <button
    class="tc__btn tc__tooltip-wrapper-anchor"
    data-selected="false"
    type="button"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Anchor
    </div>
  </button>
</div>
`;

exports[`TooltipWrapper render a simple TooltipWrapper, click to open tooltip 1`] = `
<div>
  <div
    class="tc__tooltip-wrapper"
    style="--anchorPosX: 0px; --anchorPosY: 0px;"
  >
    <div
      class="tc__tooltip-wrapper-backdrop"
    />
    <button
      class="tc__btn tc__tooltip-wrapper-anchor"
      data-selected="false"
      type="button"
    >
      <div
        style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Anchor
      </div>
    </button>
    <div
      class="tc__tooltip tc__tooltip__bottom-right"
      style="--anchorWidth: 0px;"
    >
      <div
        style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Tooltip
      </div>
    </div>
  </div>
</div>
`;

exports[`TooltipWrapper render a simple TooltipWrapper, click to open tooltip, click backdrop to close 1`] = `
<div>
  <button
    class="tc__btn tc__tooltip-wrapper-anchor"
    data-selected="false"
    type="button"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Anchor
    </div>
  </button>
</div>
`;

exports[`TooltipWrapper render a simple TooltipWrapper, mouseover to open tooltip 1`] = `
<div>
  <div
    class="tc__tooltip-wrapper"
    style="--anchorPosX: 0px; --anchorPosY: 0px;"
  >
    <div
      class="tc__tooltip-wrapper-backdrop"
    />
    <button
      class="tc__btn tc__tooltip-wrapper-anchor"
      data-selected="false"
      type="button"
    >
      <div
        style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Anchor
      </div>
    </button>
    <div
      class="tc__tooltip tc__tooltip__bottom-right"
      style="--anchorWidth: 0px;"
    >
      <div
        style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Tooltip
      </div>
    </div>
  </div>
</div>
`;

exports[`TooltipWrapper render a simple TooltipWrapper, mouseover to open tooltip, mpuseout to close 1`] = `
<div>
  <button
    class="tc__btn tc__tooltip-wrapper-anchor"
    data-selected="false"
    type="button"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Anchor
    </div>
  </button>
</div>
`;
